src/java.base/share/classes/java/text/SimpleDateFormat.java

```diff
@@ -1320,7 +1327,11 @@ private void subFormat(int patternCharIndex, int count,
             }
 
             int num = (value / 60) * 100 + (value % 60);
-            CalendarUtils.sprintf0d(buffer, num, width);
+            if (buffer.isProxyStringBuilder()) {
```

Comment:
If we made the implementations of `StringBuf` records, we could use record patterns and do away with the `isProxyStringBuilder()` method. Although we would have to change the visibility of the implementations, so maybe not...